### DISCUSSION TITLE
FFmpeg

### DISCUSSION THREAD

**Discussion by bchah** [OvenMediaLabs/OvenMediaEngine#1710] [Help] [answered]
Can OME be built without FFmpeg? OR, is it possible for end-users to replace the instance of FFmpeg used in OME with their own version? (Assuming their version supports all of the requisite functions needed by OME)

**Comment by getroot** [comment, marked as answer]
OME relies on ffmpeg for its codec interface, media file parsing (in scheduled channels), push, and recording functions. In order to build ome without ffmpeg, we need to remove these functions or improve them so that they are not dependent on it. There is a task on our backlog to remove ffmpeg dependency. It is not a big task except for media file parsing, but it is very low priority (since it does not affect the operation).

There is a great project called archlinux. They have OME in their package and it seems to use the prerequisites library installed on the system instead of /opt installed by OME. Check out this project. I believe @hashworks  contributed to it.

https://archlinux.org/packages/extra/x86_64/ovenmediaengine/

**Reply by hashworks** [reply]
To add to this: You can find the dependencies installed on Arch and the build steps in the [PKGBUILD script](https://gitlab.archlinux.org/archlinux/packaging/packages/ovenmediaengine/-/blob/main/PKGBUILD?ref_type=heads). It does indeed only use shared system libs, this was made possible by #928 - so just set `DISABLE_CUSTOM_LIBRARY_PATHS=true` and it won't use `/opt` at all. Note that the current master won't build with ffmpeg 7, for that you also need patch #1626.

**Reply by hashworks** [reply]
One downside is that the ffmpeg provided by Arch Linux isn't built with `--enable-libopenh264`, so you need to build your own ffmpeg if you need openh264 encoding in OME.
